api/layer/anonymous.go

```diff
@@ -0,0 +1,32 @@
+package layer
+
+import (
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	"github.com/nspcc-dev/neofs-sdk-go/user"
+)
+
+// Anonymous is helper struct to contain prepared data for anonymous requests.
+type Anonymous struct {
+	signer user.Signer
+	owner  user.ID
+}
+
+// NewAnonymous is a constructor for the Anonymous type.
+func NewAnonymous(pk keys.PrivateKey) *Anonymous {
+	signer := user.NewAutoIDSignerRFC6979(pk.PrivateKey)
+
+	return &Anonymous{
+		signer: signer,
+		owner:  signer.UserID(),
+	}
+}
+
+// Signer returns signer for the corresponding key.
+func (a Anonymous) Signer() user.Signer {
+	return a.signer
+}
+
+// Owner returns [user.ID] for the corresponding key.
+func (a Anonymous) Owner() user.ID {
```

Comment:
Do we need it if it's just another way to say `UserID()`? I think `Anonymous` can be replaced with `user.ID` completely.

Reply:
Replaced with user.ID